cmd/agent/main.go

```diff
@@ -185,31 +187,70 @@ func enableEdgeMode(tunnelOperator agent.TunnelOperator, clusterService agent.Cl
 				return err
 			}
 		}
+	} else {
+		log.Println("[DEBUG] [main,edge] [message: Edge key not specified. Serving Edge UI]")
+		edgeServer := http.NewEdgeServer(tunnelOperator, clusterService)
 
-		return tunnelOperator.Start()
```

Comment:
Since the tunnel operator is not started here anymore the following debug statement should be updated.

```
		log.Println("[DEBUG] [main,edge] [message: Edge key available. Starting tunnel operator.]")
```

to:

```
		log.Println("[DEBUG] [main,edge] [message: Edge key found in environment. Associating Edge key to cluster.]")
```